### PR TITLE
`azurerm_data_factory_trigger_blob_event`: Support for activation

### LINKED ISSUE #12621: Support for start/activation of Azure Data Factory blob event-based triggers
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I would like to start/activate my triggers automatically based on a flag within the `azurerm_data_factory_trigger_blob_event` resource. Also, when I delete a trigger it should deactivate/stop the trigger as it blocks my deletion actions for now.

### New or Affected Resource(s)

* `azurerm_data_factory_trigger_blob_event`

### Potential Terraform Configuration

```hcl
resource "azurerm_data_factory_trigger_blob_event" "test" {
   activated = true

   name                = "acctestdf%d"
   data_factory_id     = azurerm_data_factory.test.id
   storage_account_id  = azurerm_storage_account.test.id
   events              = ["Microsoft.Storage.BlobCreated", "Microsoft.Storage.BlobDeleted"]
   blob_path_ends_with = ".txt"
   ignore_empty_blobs  = true

   annotations = ["test1", "test2", "test3"]
   description = "test description"

   pipeline {
     name = azurerm_data_factory_pipeline.test.name
     parameters = {
       Env = "Test"
     }
   }

   additional_properties = {
     foo = "test1"
     bar = "test2"
   }
 }
```

### References

* #7574

### CC

- @njuCZ Did you leave this open on purpose (or to implement it as a separate resource)? Would you mind if I integrated it in this trigger as specified above?

### PR DESCRIPTION
**Pull request by aristosvo**
Fixes #12621

## Acceptance Tests
```
❯ make acctests SERVICE='datafactory' TESTARGS='-run=TestAccDataFactoryTriggerBlobEvent'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./azurerm/internal/services/datafactory -run=TestAccDataFactoryTriggerBlobEvent -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccDataFactoryTriggerBlobEvent_basic
=== PAUSE TestAccDataFactoryTriggerBlobEvent_basic
=== RUN   TestAccDataFactoryTriggerBlobEvent_requiresImport
=== PAUSE TestAccDataFactoryTriggerBlobEvent_requiresImport
=== RUN   TestAccDataFactoryTriggerBlobEvent_complete
=== PAUSE TestAccDataFactoryTriggerBlobEvent_complete
=== RUN   TestAccDataFactoryTriggerBlobEvent_update
=== PAUSE TestAccDataFactoryTriggerBlobEvent_update
=== RUN   TestAccDataFactoryTriggerBlobEvent_startStop
=== PAUSE TestAccDataFactoryTriggerBlobEvent_startStop
=== CONT  TestAccDataFactoryTriggerBlobEvent_basic
=== CONT  TestAccDataFactoryTriggerBlobEvent_update
=== CONT  TestAccDataFactoryTriggerBlobEvent_startStop
=== CONT  TestAccDataFactoryTriggerBlobEvent_complete
=== CONT  TestAccDataFactoryTriggerBlobEvent_requiresImport
--- PASS: TestAccDataFactoryTriggerBlobEvent_basic (173.05s)
--- PASS: TestAccDataFactoryTriggerBlobEvent_complete (182.11s)
--- PASS: TestAccDataFactoryTriggerBlobEvent_requiresImport (188.63s)
--- PASS: TestAccDataFactoryTriggerBlobEvent_startStop (264.47s)
--- PASS: TestAccDataFactoryTriggerBlobEvent_update (324.44s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/datafactory 325.861s
```

### CODE REVIEW

File: azurerm/internal/services/datafactory/data_factory_trigger_blob_event_resource.go [hunk 1]
```diff
@@ -92,6 +92,11 @@ func resourceDataFactoryTriggerBlobEvent() *pluginsdk.Resource {
 				},
 			},
 
+			"activated": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
```
Comment (tombuildsstuff) [inline azurerm/internal/services/datafactory/data_factory_trigger_blob_event_resource.go:97]: should this be defaulted to true?

File: azurerm/internal/services/datafactory/data_factory_trigger_blob_event_resource_test.go [hunk 2]
```diff
@@ -90,6 +90,30 @@ func TestAccDataFactoryTriggerBlobEvent_update(t *testing.T) {
...
+		},
+		data.ImportStep(),
+		{
+			Config: r.activated(data, false),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("activated").HasValue("false"),
+			),
+		},
+		data.ImportStep(),
```
Comment (tombuildsstuff) [inline azurerm/internal/services/datafactory/data_factory_trigger_blob_event_resource_test.go:113]: might be overkill but should we check we can update this back to true too?

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @aristosvo 

Thanks for this PR.

I've taken a look through and left a couple of minor comments inline but this otherwise LGTM 👍 

Thanks!

File: azurerm/internal/services/datafactory/data_factory_trigger_blob_event_resource.go (hunk 1 above)
Comment (aristosvo) [reply azurerm/internal/services/datafactory/data_factory_trigger_blob_event_resource.go:97]: Yes, to have best (functional) backward compatibility we probably should indeed. Consider it done!

File: azurerm/internal/services/datafactory/data_factory_trigger_blob_event_resource_test.go (hunk 2 above)
Comment (aristosvo) [reply azurerm/internal/services/datafactory/data_factory_trigger_blob_event_resource_test.go:113]: Sure, thanks!

**Comment by aristosvo** [comment]
Thanks a lot! Addressed, tests still passing:

```
❯ make acctests SERVICE='datafactory' TESTARGS='-run=TestAccDataFactoryTriggerBlobEvent'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./azurerm/internal/services/datafactory -run=TestAccDataFactoryTriggerBlobEvent -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccDataFactoryTriggerBlobEvent_basic
=== PAUSE TestAccDataFactoryTriggerBlobEvent_basic
=== RUN   TestAccDataFactoryTriggerBlobEvent_requiresImport
=== PAUSE TestAccDataFactoryTriggerBlobEvent_requiresImport
=== RUN   TestAccDataFactoryTriggerBlobEvent_complete
=== PAUSE TestAccDataFactoryTriggerBlobEvent_complete
=== RUN   TestAccDataFactoryTriggerBlobEvent_update
=== PAUSE TestAccDataFactoryTriggerBlobEvent_update
=== RUN   TestAccDataFactoryTriggerBlobEvent_startStop
=== PAUSE TestAccDataFactoryTriggerBlobEvent_startStop
=== CONT  TestAccDataFactoryTriggerBlobEvent_basic
=== CONT  TestAccDataFactoryTriggerBlobEvent_update
=== CONT  TestAccDataFactoryTriggerBlobEvent_complete
=== CONT  TestAccDataFactoryTriggerBlobEvent_requiresImport
=== CONT  TestAccDataFactoryTriggerBlobEvent_startStop
--- PASS: TestAccDataFactoryTriggerBlobEvent_basic (168.97s)
--- PASS: TestAccDataFactoryTriggerBlobEvent_requiresImport (181.14s)
--- PASS: TestAccDataFactoryTriggerBlobEvent_complete (184.45s)
--- PASS: TestAccDataFactoryTriggerBlobEvent_update (317.34s)
--- PASS: TestAccDataFactoryTriggerBlobEvent_startStop (329.12s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/datafactory   330.563s
```

**Review by katbyte — APPROVED** [review]
Thanks @aristosvo - LGTM 🌈

**Pull request merged**